tests/mocha/unit/filters/date.spec.js

```diff
@@ -15,30 +15,67 @@ describe('date filter', function() {
     state: 'STATE',
   };
 
+  const camelCaseToDash = function(string) {
+    return string.replace(/[A-Z]/g, function(match) {
+      return '-' + match[0].toLowerCase();
+    });
+  };
+
   const FormatDate = {
     age: momentDate => `${momentDate.year() - 1970} years`,
     date: d => `${d.toISOString().split('T')[0]}`,
     datetime: d => `${d.toISOString()}`,
     relative: d => `${Math.floor((d - TEST_DATE) / 86400000)} days`,
   };
+  const RelativeDate = {
+    getCssSelector: () => {
+      return 'update-relative-date';
+    },
+    generateDataset: (date, options) => {
+      let dataset = [];
+      dataset.push(`data-date="${date.valueOf()}"`);
+
+      let allowedKeys = ['age', 'withoutTime', 'end'];
+      for (let key in options) {
+        if (allowedKeys.indexOf(key) !== -1 && options[key]) {
+          dataset.push(`data-${camelCaseToDash(key)}="${options[key]}"`);
+        }
+      }
+
+      return dataset.join(' ');
+    }
+  };
```

Comment:
Instead of implementing the function I think it's better to use sinon to mock the call. You should be able to do something like...

```js
const RelativeDate = {
  generateDataset: sinon.stub()
};
RelativeDate.generateDataset.returns('data-without-time="true"');
```

Reply:
Done.